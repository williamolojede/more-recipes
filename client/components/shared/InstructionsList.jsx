import React from 'react';

const InstructionsList = () => (
  <div className="recipe__instructions col s12 m6">
    <h3>Instructions</h3>
    <ol>
      <li>
        <span>In a large bowl, sift together the flour, baking powder, salt and sugar. Make a well in the center and pour in the milk, egg and melted butter; mix until smooth.</span>
      </li>
      <li>
        Heat a lightly oiled griddle or frying pan over medium high heat. Pour or scoop the batter onto the griddle, using approximately 1/4 cup for each pancake. Brown on both sides and serve hot.
      </li>
    </ol>
  </div>
);

export default InstructionsList;
