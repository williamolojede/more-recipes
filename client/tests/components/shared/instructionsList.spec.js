import React from 'react';
import InstructionsList from '../../../components/shared/InstructionsList';
import recipesMock from '../../__mocks__/recipes.mock';

function setup() {
  const props = {
    instructions: recipesMock.recipes[1].instructions
  };
  const wrapper = mount(<InstructionsList {...props} />);

  return {
    wrapper
  };
}

describe('Instructions List', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
