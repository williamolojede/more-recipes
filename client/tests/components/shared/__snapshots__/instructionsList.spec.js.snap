// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions List should render with right amount of elements 1`] = `
<InstructionsList
  instructions={
    Array [
      "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
      "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
      "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
    ]
  }
>
  <div
    className="recipe__instructions col s12 m6"
  >
    <h3>
      Instructions
    </h3>
    <ol>
      <li
        key="0"
      >
        <span>
          Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato
        </span>
      </li>
      <li
        key="1"
      >
        <span>
          Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.
        </span>
      </li>
      <li
        key="2"
      >
        <span>
          Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.
        </span>
      </li>
    </ol>
  </div>
</InstructionsList>
`;
